src/NServiceBus.Gateway/Gateway.cs

```diff
@@ -89,7 +89,15 @@ protected override void Setup(FeatureConfigurationContext context)
             context.Pipeline.Register("GatewayIncomingBehavior", new GatewayIncomingBehavior(), "Extracts gateway related information from the incoming message");
             context.Pipeline.Register("GatewayOutgoingBehavior", new GatewayOutgoingBehavior(), "Puts gateway related information on the headers of outgoing messages");
 
-            context.RegisterStartupTask(b => new GatewayReceiverStartupTask(channelManager, channelReceiverFactory, GetEndpointRouter(context), b.Build<IDispatchMessages>(), b.Build<IDeduplicateMessages>(), b.BuildAll<IDataBus>()?.FirstOrDefault(), gatewayInputAddress));
+            context.RegisterStartupTask(b => new GatewayReceiverStartupTask(
+                channelManager, 
+                channelReceiverFactory, 
+                GetEndpointRouter(context),
+                b.Build<IDispatchMessages>(), 
+                new LegacyDeduplicationWrapper(b.Build<IDeduplicateMessages>()), 
+                b.BuildAll<IDataBus>()?.FirstOrDefault(), 
+                gatewayInputAddress,
+                context.Settings.Get<TransportInfrastructure>().TransactionMode));
```

Comment:
Should this be using https://github.com/Particular/NServiceBus/blob/ef1bcfd9cc1e9f0fa77c4e1ce499f3bce4a457b3/src/NServiceBus.Core/Transports/TransactionModeSettingsExtensions.cs instead?

Reply:
I decided not to use it for two reasons:
* It throws an exception on send-only endpoints and I'd have to catch that
* It uses the actually configured transaction mode, which is something we should do indeed but not what the gateway is doing currently. I'd like to handle that case in a dedicated PR.

Reply:
Got it